src/test_cases.py

```diff
@@ -0,0 +1,170 @@
+
+    # MONAI
+    metatensor = image_to_metatensor(image)
+    affine_matrix_for_monai = itk_to_monai_affine(image, matrix=matrix, translation=translation)
+    output_array_monai = transform_affinely_with_monai(metatensor, affine_matrix=affine_matrix_for_monai)
+
+    # Transformix
+    output_array_transformix = transform_affinely_with_transformix(image, matrix=matrix, translation=translation)
+
+    ###########################################################################
+    # Make sure that the array conversion of the inputs is the same 
+    input_array_monai = metatensor_to_array(metatensor)
+    assert(np.array_equal(input_array_monai, np.asarray(image)))
+
+    # Compare outputs
+    print("MONAI-ITK: ", np.allclose(output_array_monai, output_array_itk))
+    print("ITK-Transformix: ", np.allclose(output_array_itk, output_array_transformix))
+
+    diff_output = output_array_monai - output_array_itk
+    print("[Min, Max] MONAI: [{}, {}]".format(output_array_monai.min(), output_array_monai.max()))
+    print("[Min, Max] ITK: [{}, {}]".format(output_array_itk.min(), output_array_itk.max()))
+    print("[Min, Max] Transformix: [{}, {}]".format(output_array_transformix.min(), output_array_transformix.max()))
+    print("[Min, Max] diff: [{}, {}]".format(diff_output.min(), diff_output.max()))
+
+    # Write 
+    # itk.imwrite(itk.GetImageFromArray(diff_output), "./output/diff.tif")
+    # itk.imwrite(itk.GetImageFromArray(output_array_monai), "./output/output_monai.tif")
+    # itk.imwrite(itk.GetImageFromArray(output_array_itk), "./output/output_itk.tif")
+    # itk.imwrite(itk.GetImageFromArray(output_array_transformix), "./output/output_transformix.tif")
+    ###########################################################################
+
+
+
+def test_arbitary_center_of_rotation(filepath):
+    print("\nTEST: affine matrix with arbitary center of rotation")
+    # Read image
+    image = itk.imread(filepath, itk.F)
+    image[:] = remove_border(image)
+    ndim = image.ndim
+
+    # ITK matrix (3x3 affine matrix)
+    matrix = np.array([[0.55915995, 0.50344867, 0.43208387],
+                       [0.01133669, 0.82088571, 0.86841365],
+                       [0.30478496, 0.94998986, 0.32742505]])[:ndim, :ndim]
+    translation = [54.0, 2.7, -11.9][:ndim]
+
+    # Spatial properties
+    center_of_rotation = [-32.3, 125.1, 0.7][:ndim]
+    origin = [1.6, 0.5, 2.0][:ndim]
+    spacing = np.array([1.2, 1.5, 0.6])[:ndim]
+
+    image.SetSpacing(spacing)
+    image.SetOrigin(origin)
+
+    # ITK
+    output_array_itk = transform_affinely_with_itk(image, matrix=matrix, translation=translation, center_of_rotation=center_of_rotation)
+
+    # MONAI
+    metatensor = image_to_metatensor(image)
+    affine_matrix_for_monai = itk_to_monai_affine(image, matrix=matrix, translation=translation, center_of_rotation=center_of_rotation)
+    output_array_monai = transform_affinely_with_monai(metatensor, affine_matrix=affine_matrix_for_monai)
```

Comment:
Naming suggestion:

transform_affinely_with_monai -> monai_affine_resample or affine_resample if in the monai package